model/textparse/openmetricsparse.go

```diff
@@ -222,9 +222,82 @@ func (p *OpenMetricsParser) Exemplar(e *exemplar.Exemplar) bool {
 // CreatedTimestamp returns nil as it's not implemented yet.
 // TODO(bwplotka): https://github.com/prometheus/prometheus/issues/12980
 func (p *OpenMetricsParser) CreatedTimestamp() *int64 {
+	newParser := deepCopyParser(p)
+	switch t, _ := newParser.Next(); t {
+		case EntrySeries:
+			for {
+				// Check _created suffix
+				var lbs labels.Labels
+				newParser.Metric(&lbs)
+				name := lbs.Get(model.MetricNameLabel)
+				if name[len(name)-8:] != "_created" {
+					return nil
+				}
+
+				var labelsPrev labels.Labels
+				p.Metric(&labelsPrev) 
+				namePrev := labelsPrev.Get(model.MetricNameLabel)
+				// a case like foo_total != foo_created fails even if underlying metric is the same
+				if namePrev != name { 
+					return nil
+				}
+
```

Comment:
Another TODO; the label set must be the same. While unlikely since interleaving is not allowed (https://github.com/OpenObservability/OpenMetrics/blob/main/specification/OpenMetrics.md#metricpoint-1), it is possible and we should detect that error.

Reply:
same as above